Do not assign a y position to notes

Navid200 opened the pull request:
Fixes: https://github.com/NightscoutFoundation/xDrip/issues/1299  
<br/>  
  
---  
  
**Recreate the problem**  
Install the latest Nightly (https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.08.07).  
Disable predictive simulations.  
Set the BG units to mmol/L.
Enter an insulin treatment.
Tap on it and add a note.  You should see something like this.  
![Screenshot_20240812-175734](https://github.com/user-attachments/assets/8b54318d-c531-44b4-a059-c0a7ae86bdd9)
  
Change the BG unit setting to mg/dL.  Now, you will see something like this.  
![Screenshot_20240812-180000](https://github.com/user-attachments/assets/e4725b3f-4889-483e-8903-9698b7872880)
  
You can see the y axis range has been expanded to very low values.  Now, delete the treatment.  You will see the scale going back to normal as shown below.
![Screenshot_20240812-180204](https://github.com/user-attachments/assets/f645260b-1055-4ce7-aa92-e93e8988c424)  
 
Enter a treatment again.  Add a note to it.  You should see something like this:  
![Screenshot_20240812-180331](https://github.com/user-attachments/assets/fb536958-8255-4aef-b9c9-bfdafbe54e77)  
Change the BG unit to mmol/L.  You will see this:  
![Screenshot_20240812-180522](https://github.com/user-attachments/assets/7dda3a63-b3d1-4036-99e6-366031ac8f7f)  
You can see that the y axis range is now unreasonably extended on the positive side.  
 <br/>  
  
---   
  
**After this PR**  
The note will not have a y position value applied to them.  
Let's repeat the second test.  This is what you will see after entering a treatment and adding a note to it.  
![Screenshot_20240812-181019](https://github.com/user-attachments/assets/459c7e7b-761c-4a1e-a7a3-1d9d8c52f6cc)  
Now, if you change the BG unit, you will see this.  
![Screenshot_20240812-181152](https://github.com/user-attachments/assets/0a28f225-38b9-4050-8d4e-f136980514a6)  
You can see that the y axis range remains reasonable.

Navid200 commented:
What gave me the idea that this was the solution was the existing note feature.
You can add a note to a treatment.  You can also just place a note.
If you look at the code, you will see that the existing code for placing a note, which does not exhibit this problem, does not assign a y position to the note by setting it to -1.

Seeing that and knowing that note does not have the issue, I tried it with treatment note and tests showed that removing the y position from treatment note solved the problem for treatment note as well.

jamorham commented:
I'm not too sure about this because it looks like you're removing the facility of notes being placed at particular points on the Y axis when the problem is that this position is encoded in the current units and if those change then the note positions become invalid.

I feel like a better solution would be to detect notes which are outside of the currently selected unit range and scale those appropriately or to convert the Y positions in the database during a units change. I think the first option is likely preferable, as for example mmol would always be under 24 and mgdl always above it.

Navid200 commented:
Thanks for the review.

But, why does this problem never occur for a note?  It only occurs for a note attached to a treatment.
The only difference is that a note is set up with -1 as the parameter.
https://github.com/NightscoutFoundation/xDrip/blob/aee9315b237d735e58434cccecdd19ef5c397a4d/app/src/main/java/com/eveningoutpost/dexdrip/Home.java#L3494  

![Screenshot 2024-09-27 105338](https://github.com/user-attachments/assets/209eb003-0c09-4a2d-9d43-a1c97d734daa)

When I saw that this problem only occurred when I attach a note to a treatment, but does not when I just enter a note, I decided to see what the difference was.
The only difference is that we enter a y position for a treatment attached to a treatment.  While we leave it blank when we only create a note.

Navid200 commented:
@jamorham Please see my post above.  This PR only does what xDrip does already for notes.  It's already been in xDrip for years and tested.  Why can we not use it for notes attached to treatments as well?

Navid200 commented:
@jamorham I can do what you have suggested.

But,  will you do me a favor?   Would you please tell me that you are aware that note (not treatment note) already does what this PR suggests and knowing that, you still prefer that I do what you have suggested?
I would just like to make sure you are aware that we are already doing what this PR suggests and we have been doing it for years.

Thanks

Navid200 commented:
Installed the latest Nightly, 2024.09.27, on a test phone.  Set the units to mg/dL.  Added a note.
![Screenshot_20240930-170953](https://github.com/user-attachments/assets/c4283ab0-5101-4faf-894e-7e2df1f673db)
<br/>  

Added a treatment.  
![Screenshot_20240930-171517](https://github.com/user-attachments/assets/1060b67d-3842-4c64-9e3a-075928c03f6c)
<br/>  

Changed the unit to mmol/L.  The y axis range remains unchanged.  
![Screenshot_20240930-171545](https://github.com/user-attachments/assets/f8515073-5b3c-41ff-adc3-26fa7d5ed8ca)
<br/>  
Changed the unit back to mg/dL.  Tapped on the treatment and added a note (Test2) to it.
![Screenshot_20240930-171641](https://github.com/user-attachments/assets/d5a87043-c5ab-4679-b00a-9ba4c67e650b)
<br/>  
Changed the unit to mmol/L.  
![Screenshot_20240930-171715](https://github.com/user-attachments/assets/5b38ee00-bd9d-49dc-b7ed-494f5c543c60)
<br/>  
  
You can see that the note remains where it was.  But, the treatment is now at a very high value causing the y axis range to expand.

Navid200 commented:
> I'm not too sure about this because it looks like you're removing the facility of notes being placed at particular points on the Y axis when the problem is that this position is encoded in the current units and if those change then the note positions become invalid.  
  
The y position is defined by the treatment.  The note added to the treatment overrides that position in xDrip as is.  This PR avoids impacting the position of the treatment when adding a note to it.

Navid200 commented:
@jamorham Why should we allow the position of a treatment to be changed just because a note has been added to it?
This PR places the note where the treatment is.  If the unit is changed then, the treatment moves accordingly as it did before there was a note attached to it.  The note has no choice but to move with it.

Would you please reconsider?
